tensorboard/defs/web.bzl

```diff
@@ -377,3 +379,60 @@ tf_web_library = rule(
         "_closure_library_base": CLOSURE_LIBRARY_BASE_ATTR,
     }.items()),
     outputs=DEPRECATED_CLUTZ_OUTPUTS)
+
+def _tf_web_test_python_shell(ctx):
+  ctx.actions.expand_template(
+      template=ctx.file._template,
+      output=ctx.outputs.main,
+      substitutions={
+          "{BINARY_PATH}": ctx.executable.web_library.short_path,
+          "{WEB_PATH}": ctx.attr.web_path,
+      }
+  )
+
+tf_web_test_python_shell = rule(
+    implementation=_tf_web_test_python_shell,
+    attrs={
+        "web_library": attr.label(
+            executable=True,
+            cfg="host",
+            mandatory=True,
+            doc="A tf_web_library target that serves the test suite",
+        ),
+        "web_path": attr.string(
+            mandatory=True,
+            doc="The web path for the HTML entry point. This will be " +
+                "the concatenation of the tf_web_library's path with " +
+                "the name of the main HTML source file: for example, " +
+                "\"/vz-foo/test/test.html\".",
+        ),
+        "main": attr.output(),
+        "_template": attr.label(
+            default=Label("//tensorboard/defs:web_test_python_shell.template.py"),
+            allow_single_file=True,
+        ),
+    },
+)
+
+def tf_web_test(name, web_library, web_path, **kwargs):
```

Comment:
use kawrgs or just omit it

Reply:
I did this but forgot to fold it into the commit. Will push presently.